src/phoenix/db/database.py

```diff
@@ -51,24 +62,43 @@
     events JSON,
     status TEXT CHECK(status IN ('UNSET','OK','ERROR')) NOT NULL DEFAULT('UNSET'),
     status_message TEXT,
+    latency_ms REAL,
     cumulative_error_count INTEGER NOT NULL DEFAULT 0,
     cumulative_llm_token_count_prompt INTEGER NOT NULL DEFAULT 0,
     cumulative_llm_token_count_completion INTEGER NOT NULL DEFAULT 0,
-    FOREIGN KEY(trace_rowid) REFERENCES traces(rowid)
+    FOREIGN KEY(trace_rowid) REFERENCES traces(id)
 );
 CREATE INDEX idx_parent_span_id ON spans(parent_span_id);
 PRAGMA user_version = 1;
 COMMIT;
 """
 
 
+_MEM_DB_STR = "file::memory:?cache=shared"
+
+
+def _mem_db_creator() -> Any:
+    return sqlite3.connect(_MEM_DB_STR, uri=True)
+
+
+@event.listens_for(Engine, "connect")
+def set_sqlite_pragma(dbapi_connection: Connection, _: Any) -> None:
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA foreign_keys = ON;")
+    cursor.execute("PRAGMA journal_mode = WAL;")
+    cursor.execute("PRAGMA synchronous = OFF;")
+    cursor.execute("PRAGMA cache_size = -32000;")
+    cursor.execute("PRAGMA busy_timeout = 10000;")
+    cursor.close()
```

Comment:
oh interesting. Does this need to happen every time or can it be done on the first migration?

Reply:
The first two are persisted (i think). But the rest are run-time configs.